concepts/05 UI Components/DataGrid/20 Editing/43 Customize Edit Form/30 Save, Cancel, and Custom Buttons.md

```diff
@@ -0,0 +1,23 @@
+Form and popup editing modes include built-in Save and Cancel buttons. If you want to execute custom code on click one of these buttons, specify the [onSaving](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onSaving) or [onEditCanceling](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onEditCanceling) event handler correspondingly. Assign `true` to the **e.cancel** parameter to cancel the Save or Cancel operations and keep the form or popup opened.
+
+Specify the **editing.texts**.[cancelRowChanges](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/texts/#cancelRowChanges) and **editing.texts**.[saveRowChanges](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/texts/#saveRowChanges) properties to change texts of the Save and Cancel buttons.
+
+To customize the Save and Cancel buttons, re-define them:
+
+- Popup edit mode     
+Add the **editing**.[popup](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/#popup) object to the DataGrid configuration and declare the Save, Cancel and custom buttons in the **popup**.[toolbarItems](/Documentation/ApiReference/UI_Components/dxPopup/Configuration/toolbarItems/) array.
+
+- Form edit mode     
+Add the **editing**.[form](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/#form) object to the DataGrid configuration and use the **form.items** array to arrange all items you want to display in the edit form. Then, add a [simple item](/Documentation/ApiReference/UI_Components/dxForm/Item_Types/SimpleItem/) with a template containing a standalone [Toolbar](/Documentation/Guide/UI_Components/Toolbar/Getting_Started_with_Toolbar/) component. Declare the buttons in this toolbar. Use the following CSS to hide the built-in Save and Cancel buttons:
```

Comment:
>  Declare **the** buttons in this toolbar.

You say "the buttons". I ask -- what buttons? If I cannot find an answer, I change the definite case to indefinite.

> to hide **the** built-in Save and Cancel buttons:

I understand that you considered that "built-in" adds definiteness, akin to the word "default". Yes, we probably may interpret the scenario this way. However, in this is not important here. To do what? To hide some built-in Save and Cancel buttons, for there can be many of them. We don't know, and it's not important in the current context. I wouldn't called this an error, but our current style gravitates towards zero article in such cases.

```suggestion
Add the **editing**.[form](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/#form) object to the DataGrid configuration and use the **form.items** array to arrange all items you want to display in the edit form. Then, add a [simple item](/Documentation/ApiReference/UI_Components/dxForm/Item_Types/SimpleItem/) with a template containing a standalone [Toolbar](/Documentation/Guide/UI_Components/Toolbar/Getting_Started_with_Toolbar/) component. Declare buttons in this toolbar. Use the following CSS to hide built-in Save and Cancel buttons:
```

P.S. The things I mentioned in the previous edit, also apply to this paragraph as well.